Clean up policy sets page (#516)
hashicorp/terraform-docs-common: website/docs/cloud-docs/policy-enforcement/manage-policy-sets.mdx

To view and manage policies and policy sets, you must have [manage policy permissions](/terraform/cloud-docs/users-teams-organizations/permissions#manage-policies) for your organization.

## Comparing policy checks and evaluations

> Sentinel policy sets can run as both policy checks and evaluations, while OPA policy sets only run as evaluations. Policy checks and evaluations can access different types of data and enable slightly different workflows.

Reviewer: > Sentinel policy sets can run as both policy checks and evaluations, while OPA policy sets only run as evaluations

Is this required since we already call this out in both the **Policy Checks** and **Policy Evaluations** sections below.

> Sentinel policy sets can run as both policy checks and evaluations, while OPA policy sets only run as evaluations. Policy checks and evaluations can access different types of data and enable slightly different workflows.

Author: ```suggestion
Policy checks and evaluations can access different types of data and enable slightly different workflows.
```

-> **Note:** Sentinel policy evaluations are currently in beta.
Sentinel policies can run as a policy evaluation on an opt-in basis.

## Management Workflows
For Sentinel policy sets, using policy evaluations lets you:
- Enable overrides for soft-mandatory and hard-mandatory policies, letting any user with [Manage Policy Overrides permission](/terraform/cloud-docs/users-teams-organizations/permissions#manage-policy-overrides) proceed with a run in the event of policy failure.
- Select a specific Sentinel runtime version for the policy set.

You can use the following workflows to manage policies and policy sets for your Terraform Cloud organization:
- **Individually Managed:** Add policies directly in the Terraform Cloud UI, and Terraform Cloud stores your policy code. This workflow is ideal for initial experimentation with policy enforcement, but we do not recommend it for organizations with large numbers of policies.
- **Version Control:**  Connect Terraform Cloud to a version control repository containing a policy set. When you push changes to the repository, Terraform Cloud automatically uses the updated policy set.

> - **Automated:** Push versions of policy sets to Terraform Cloud with the [Terraform Cloud Policy Sets API](/terraform/cloud-docs/api-docs/policy-sets#create-a-policy-set-version). For Sentinel only, you can also use the  `tfe` provider [`tfe_policy_set`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/policy_set) resource. This workflow is ideal for automated Continuous Integration and Deployment (CI/CD) pipelines.

Reviewer: > For Sentinel only, you can also use the  `tfe` provider [`tfe_policy_set`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/policy_set) resource.

@mrinalirao I just want to double check the above with you as I was under the impression that we support both Sentinel and OPA.

Reviewer 2: yes, the docs seem to be incorrect, we support both Sentinel and OPA for tfe_policy_set provider

- **advisory** Failed policies never interrupt the run. They provide information about policy failures in the UI.
- **mandatory:** Failed policies stop the run, but any user with [Manage Policy Overrides permission](/terraform/cloud-docs/users-teams-organizations/permissions#manage-policy-overrides) can override these failures and allow the run to complete.

## Policy publishing workflows

You can create policies and policy sets for your Terraform Cloud organization in one of three ways:
- **Terraform Cloud web UI:** Add individually-managed policies manually in the Terraform Cloud UI, and store your policy code in Terraform Cloud. This workflow is ideal for initial experimentation with policy enforcement, but we do not recommend it for organizations with large numbers of policies.
- **Version control:**  Connect Terraform Cloud to a version control repository containing a policy set. When you push changes to the repository, Terraform Cloud automatically uses the updated policy set.

> - **Automated:** Push versions of policy sets to Terraform Cloud with the [Terraform Cloud Policy Sets API](/terraform/cloud-docs/api-docs/policy-sets#create-a-policy-set-version). For Sentinel only, you can also use the  `tfe` provider [`tfe_policy_set`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/policy_set) resource. This workflow is ideal for automated Continuous Integration and Deployment (CI/CD) pipelines.

Author: ```suggestion
- **Automated:** Push versions of policy sets to Terraform Cloud with the [Terraform Cloud Policy Sets API](/terraform/cloud-docs/api-docs/policy-sets#create-a-policy-set-version) or the  `tfe` provider [`tfe_policy_set`](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/policy_set) resource. This workflow is ideal for automated Continuous Integration and Deployment (CI/CD) pipelines.
```

1. **(VCS Only)** Specify where your policy set files live using the **Policies path**. This lets you maintain multiple policy sets within a single repository. Use a relative path from your root directory to the directory that contains either the `sentinel.hcl` (Sentinel) or `policies.hcl` (OPA) configuration files. If you do not set this field, Terraform Cloud uses the repository's root directory.
1. **(Managed Policies Only)** Select managed **Policies** to add to the policy set. You can only add policies written with the same policy framework you selected for this set.
1. Choose a descriptive and unique **Name** for the policy set. You can use any combination of letters, numbers, `-`, and `_`.
1. Write an optional **Description** that tells other users about the purpose of the policy set and what it contains.

### Edit policy sets

To edit a policy set:
1. Go to the **Policy Sets** section of your organization’s settings.
1. Click the policy set you want to edit to go to its settings page.
1. Adjust the settings and click **Update policy set**.

> ### Evaluate a policy upgrade

Reviewer: ```suggestion
### Evaluate a policy runtime upgrade
```
@ritsok I think we should explicitly call out "runtime" in the section title

You can validate that changing a policy runtime version does not introduce any breaking changes.

-> **Note:** On-demand policy evaluation is in beta.

To perform a policy evaluation:

1. Go to the **Policy Sets** section of your organization’s settings.
1. Click the policy set you want to upgrade.
1. Click the **Evaluate** tab.
1. Select the **Runtime version** you wish to upgrade to.
1. Select a **Workspace** to test the policy and upgraded version against.
1. Click **Evaluate**.

> Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation fails, FILL HERE.

Author: @hcrhall or @cam-stitt: what guidance should we offer users here?

Reviewer: @cam-stitt correct me if I am wrong but the status if either `passed` or `errored`. 

I think we should say something like: 

```suggestion
Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. The evaluation results will be displayed. If the evaluation has an 'Errored' status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue within a policy.
```

Reviewer 3: There are three statuses that can happen. `passed` when everything is fine. `failed` if a policy returned false and `errored` if the entire process died and nothing could happen. It it reaches `errored`, no JSON will be displayed because no policies were even run.

![Screenshot 2023-12-13 at 9 07 15 am](https://github.com/hashicorp/terraform-docs-common/assets/186643/d616a464-a730-4bc9-8046-1b671a799e64)

Author: is there any action a user can/should take in the event of an error?

Reviewer 3: Other than notifying an administrator or contacting support if they believe the policy set should have run successfully, no. It means that someone stopped the policy set from even running, such as invalid policy set configuration.

> Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation fails, FILL HERE.

Author: ```suggestion
Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation has an `Errored` status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue within a policy.
```

> Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation has an `Errored` status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue within a policy.

Author: ```suggestion
Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation returns a `Failed` status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue within a policy.
```

> Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation returns a `Failed` status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue within a policy.

Author: ```suggestion
Terraform Cloud will execute the policy set using the specified version and the latest plan data for the selected workspace. It will display the evaluation results. If the evaluation returns a `Failed` status, inspect the JSON output to determine whether the issue is related to a non-compliant resource or is due to a syntax issue.
If the evaluation results in an error, check that the policy configuration is valid.
```
